src/dmd/root/outbuffer.d

```diff
@@ -386,6 +386,22 @@ struct OutBuffer
         writestring(unsignedToTempString(u, buf));
     }
 
+    /**************************************
+     * Convert `u` to a hex string and append it to the buffer.
+     * Params:
+     *  u = integral value to append
+     *  mw = min width with zero padding (e.g. u=255, mw=8 : '000000ff')
+     */
+    extern (C++) void printHex(ulong u, uint mw = 0) pure nothrow
```

Comment:
If you're going to the trouble of exporting the function to C++, then might as well put it in the headers and add test for it to ensure it can be linked to.

Reply:
I'll add it.

Reply:
See the following unittest.
https://github.com/dlang/dmd/blob/04bf3325a3855f622d047d9317dd7a64a70f3956/src/tests/cxxfrontend.c#L434-L451

Reply:
So there are the C/C++ tests, I'll put some lines when I can.
Thanks.

Reply:
I'm saying this because I expect linking to fail, cppmangle.d does not have a direct mapping of D long to C++, so you'll need to add a compatibility type (`d_long`) to compat.h, and use that in outbuffer.h

Reply:
Do C11 types work?

Reply:
On the D side? They are just aliases to D `long`.  Mangling of long is determined by this condition:
https://github.com/dlang/dmd/blob/04bf3325a3855f622d047d9317dd7a64a70f3956/src/dmd/cppmangle.d#L1643-L1647

There is a very good reason why we have this unwieldy `#ifdef` in globals.h
https://github.com/dlang/dmd/blob/master/src/dmd/globals.h#L327-L345

Reply:
I see now, the mapping works on linux (`uint64_t` <- `ulong`) though but it seems OSX is tricky.
There is `core.stdc.stdint: uint64_t` that should work, `dmd/globals.d` is using it (`alias d_uns64 = uint64_t;`).

Reply:
code moved to #11563